packages/ti_opencti/data_stream/indicator/elasticsearch/ingest_pipeline/ecs_from_artifact.yml

```
@@ -48,13 +48,19 @@ processors:
       ignore_empty_value: true
 
   # append object
-  - append:
-      field: threat.indicator.file
-      value: []
   - script:
       description: Append to the destination
       lang: painless
       source: |
+        if (ctx.threat == null) {
+          ctx.threat = [:];
+        }
+        if (ctx.threat.indicator == null) {
+          ctx.threat.indicator = [:];
+        }
+        if (ctx.threat.indicator.file == null) {
+          ctx.threat.indicator.file = [];
+        }
```

This can be less verbose:

```suggestion
        ctx.threat = ctx.threat ?: [:];
        ctx.threat.indicator = ctx.threat.indicator ?: [:];
        ctx.threat.indicator.file = ctx.threat.indicator.file ?: [];
```

Also below.